orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -394,36 +407,65 @@ def _get_exported_model(self, input_schema, *inputs, **kwargs):
         # correct training flag to reflect the expected behavior.
         # For example, the Dropout node in a model is dropped under eval mode.
         assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
-
-        try:
-            with torch.set_grad_enabled(self._enable_custom_autograd_function), \
-                    _logger.suppress_os_stream_output(log_level=self._debug_options.logging.log_level):
-                required_export_kwargs = {'input_names': self._input_info.names,
-                                          'output_names': output_names,
-                                          'opset_version': ortmodule.ONNX_OPSET_VERSION,
-                                          'do_constant_folding': False,
-                                          'training': self._export_mode,
-                                          'dynamic_axes': self._input_info.dynamic_axes,
-                                          'verbose': self._debug_options.logging.log_level < LogLevel.WARNING,
-                                          'export_params': False,
-                                          'keep_initializers_as_inputs': True}
-                invalid_args = self._export_extra_kwargs.keys() & required_export_kwargs.keys()
-                assert len(invalid_args) == 0,\
-                    f"The following PyTorch exporter arguments cannot be specified: '{invalid_args}'."
-                torch.onnx.export(self._flattened_module,
-                                  sample_inputs_as_tuple,
-                                  f,
-                                  **required_export_kwargs,
-                                  **self._export_extra_kwargs)
-        except Exception as e:
-            raise wrap_exception(ORTModuleONNXModelException,
-                                 RuntimeError(f'There was an error while exporting the PyTorch model to ONNX: '
-                                              f'\n\n{_utils.get_exception_as_string(e)}'))
-        exported_model = onnx.load_model_from_string(f.getvalue())
-
-        exported_model = _post_process_after_export(exported_model,
+        provider = self._provider_configs.provider
+        if not provider == "openvino":
```

Comment:
Could you clarify why we need to branch this part?

Reply:
@thiagocrepaldi, we are branching out to enable changes in export params specific to inference using openvino and avoid changes to existing path specific to ORT training that includes gradient related code.

Reply:
The problem with this approach is that user does not know the implications of choosing openvino execution provider. When user chooses openvino, internally we run inference on the model which is very different than the usual flow

Reply:
may i understand what kind of exporter params specific you need? is it because of openvino or is it because you only run inference? i am feeling we are mixed those two concepts. This OpenVino PR actually introduce two things: 1) add openvino provider option. 2) implicitly add a "inference mode" in ortmodule, for example, export all the weights as initializers, not inputs. 

If we really want to do it, we should seperate them into two tasks. one is pure enable openvino provider option. another is to provide "inference mode". The "inference mode" could be used for not only openvino, but also ort CPU/Cuda/TensorRT EPs. and we should have explicit check to avoid pytorch user accidently trigger training feature on "inference mode"

Reply:
"we should have explicit check to avoid pytorch user accidently trigger training feature on "inference mode" very true!

Reply:
Export params specific to inference that are needed:
'do_constant_folding': True,
'operator_export_type': OperatorExportTypes.ONNX_ATEN_FALLBACK,
'export_params': True,
'keep_initializers_as_inputs': False

Since these params are not specific to openvino, we can have "inference mode" for more generic usage across other ort EPs and avoid accidental trigger by pytorch user. We will update the PR shortly for these changes.